cdap-sentry/cdap-sentry-extension/cdap-sentry-binding/src/test/java/co/cask/cdap/security/authorization/sentry/binding/SentryAuthorizerTest.java

```diff
@@ -503,12 +504,28 @@ public void testUser2Ns3() throws Exception {
     assertUnauthorized(new ProgramId("ns3", "a1", ProgramType.SPARK, "s1"), user, Action.EXECUTE);
 
     // dataset modules
-    assertAuthorized(new DatasetModuleId("ns3", "tab_1_mod"), user, Action.READ);
-    assertAuthorized(new DatasetModuleId("ns3", "tab_data_mod"), user, Action.WRITE);
-    assertAuthorized(new DatasetModuleId("ns3", "tab_da_4_ta_mod"), user, Action.ADMIN);
+    assertAuthorized(new DatasetModuleId("ns3", "co.cask.tab_1_mod"), user, Action.READ);
+    assertAuthorized(new DatasetModuleId("ns3", "co.cask.tab_data_mod"), user, Action.WRITE);
+    assertAuthorized(new DatasetModuleId("ns3", "co.cask.tab_da_4_ta_mod"), user, Action.ADMIN);
 
-    assertUnauthorized(new DatasetModuleId("ns2", "tab_1_mod"), user, Action.ADMIN);
-    assertUnauthorized(new DatasetModuleId("ns3", "table_1_mod"), user, Action.ADMIN);
+    assertUnauthorized(new DatasetModuleId("ns2", "co.cask.tab_1_mod"), user, Action.ADMIN);
+    assertUnauthorized(new DatasetModuleId("ns3", "co.cask.table_1_mod"), user, Action.ADMIN);
+  }
+
+  @Test
+  public void testRoleGrant() throws Exception {
+    String allPrograms = "program:gold.new_pipeline.*";
+    Principal execRole = new Principal("exec-role", Principal.PrincipalType.ROLE);
+    authorizer.createRole(new Role(execRole.getName()));
```

Comment:
Extract `new Role(execRole.getName())` so that we do not need to create it 3 times?